### PR TITLE
Support clear bot history on new build

### PR DESCRIPTION
**Pull request by devharis**
As per [this](https://github.com/googleforgames/quilkin/issues/266) issue - impl. support to clear bot history on new builds to prevent PR comment bloat.

### CODE REVIEW

**Comment by devharis** [comment]
I saw your discussion regarding hiding old comments, so you really need to preserve build history comments? If not - this would support deleting them.

**Comment by markmandel** [comment]
I didn't think to just delete the old comments on each run (rather than minimise), but it's not a bad idea to at least test.

I do like being able to see the history over time (so prefer minimisation), it can also make it easier to identify flaky tests, and I have some vague concerns about race conditions when builds are simultaneous, but I don't see any reason not to at least test it out.

I wonder if we'll hit API request limits in the beginning 😄 

@XAMPPRocky did you have any thoughts? It was you who brought up removing/hiding the old test results, so wanted to check in first before deploying anything new.

**Comment by XAMPPRocky** [comment]
It seems fine to me. My one concern about deleting, is that it would make it harder to look at old builds, which sometimes you want to do, but I'm not against deleting either.

**Comment by markmandel** [comment]
> It seems fine to me. My one concern about deleting, is that it would make it harder to look at old builds, which sometimes you want to do, but I'm not against deleting either.

Yeah, same here. But let's try it! If we hate it, we can change it again. I'll get it built and deployed, and will confirm it works on this PR, and assuming it all works, we can roll it out and see how we go!

**Comment by markmandel** [comment]
Haven't forgotten at this, just been at 100% bandwidth. Do want to get this live so we can take it for a spin, but would like to do it at a time so I can watch in case it goes badly for any reason 😄

File: build/ci/github-bot/main.go [hunk 1]
```diff
@@ -129,6 +132,22 @@ func (g *githubNotifier) SendNotification(ctx context.Context, build *cloudbuild
 	return nil
 }
 
+func (g *githubNotifier) clearBotComments(ctx context.Context, pr int) error {
+	comments, _, err := g.client.Issues.ListComments(ctx, owner, repo, pr, &github.IssueListCommentsOptions{})
+	if err != nil {
+		log.Errorf("Error retriving comment history: %v on PR with id: %s", err, pr)
```
Comment (markmandel) [inline build/ci/github-bot/main.go:138]: Small spelling error: 
```suggestion
		log.Errorf("Error retrieving comment history: %v on PR with id: %s", err, pr)
```

**Comment by markmandel** [comment]
Deploying now! Let's see how we go!

**Comment by markmandel** [comment]
Looks like it worked! Let's let it bake for a few days (also would love to get that typo fixed), and if it's going swimmingly, let's merge 👍🏻

**Review by markmandel — APPROVED** [review]

**Comment by markmandel** [comment]
Let's merge it - it's looking good 👍🏻

**Pull request merged**